test/test_train_mp_mnist.py

```diff
@@ -96,29 +98,35 @@ def train_mnist():
           shuffle=True)
     train_loader = torch.utils.data.DataLoader(
         train_dataset,
-        batch_size=FLAGS.batch_size,
+        batch_size=flags.batch_size,
         sampler=train_sampler,
-        drop_last=FLAGS.drop_last,
+        drop_last=flags.drop_last,
         shuffle=False if train_sampler else True,
-        num_workers=FLAGS.num_workers)
+        num_workers=flags.num_workers)
     test_loader = torch.utils.data.DataLoader(
         test_dataset,
-        batch_size=FLAGS.batch_size,
-        drop_last=FLAGS.drop_last,
+        batch_size=flags.batch_size,
+        drop_last=flags.drop_last,
         shuffle=False,
-        num_workers=FLAGS.num_workers)
+        num_workers=flags.num_workers)
 
   # Scale learning rate to num cores
-  lr = FLAGS.lr * xm.xrt_world_size()
+  lr = flags.lr * xm.xrt_world_size()
 
   device = xm.xla_device()
   model = MNIST().to(device)
   writer = None
   if xm.is_master_ordinal():
-    writer = test_utils.get_summary_writer(FLAGS.logdir)
-  optimizer = optim.SGD(model.parameters(), lr=lr, momentum=FLAGS.momentum)
+    writer = test_utils.get_summary_writer(flags.logdir)
+  optimizer = optim.SGD(model.parameters(), lr=lr, momentum=flags.momentum)
   loss_fn = nn.NLLLoss()
 
+  # Start up client side profiler server.
+  server = xp.start_server(flags.port)
```

Comment:
Probably `start_server` should be a no-op if port is invalid or None. Is there anything else in this file that would break if `start_server` was a no-op?

Reply:
The argument validation is handled by the c++ layer bindings: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/profiler/rpc/profiler_server.cc#L31 so IMO it'd be better to keep validation code in one place.

Reply:
SGTM. How is the crash behavior for the user if it happens in the C++ vs Python?  When we get the huge XLA graph dumps, it's pretty much a guarantee that the user won't find the error message. So I think C++ checking is fine unless the error is hard to read, in which case I'd vote for finding the issue early in the Python layer

Reply:
Ex:
```
RuntimeError: Could not interpret "localhost:-1" as a host-port pair.
```